test/e2e/specs/site-editor/navigation-editor.spec.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Editing Navigation Menus', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllMenus();
+	} );
+
+	test( 'it should lock the root Navigation block in the editor', async ( {
+		admin,
+		page,
+		pageUtils,
+		requestUtils,
+		editor,
+	} ) => {
+		// We could Navigate directly to editing the Navigation Menu but we intentionally do not do this.
+		//
+		// Why? To provide coverage for a bug that caused the Navigation Editor behaviours to fail
+		// only when navigating through the editor screens (rather than going directly to the editor by URL).
+		// See: https://github.com/WordPress/gutenberg/pull/56856.
+		//
+		// Example (what we could do):
+		// await admin.visitSiteEditor( {
+		// 	postId: createdMenu?.id,
+		// 	postType: 'wp_navigation',
+		// } );
+		//
+		await admin.visitSiteEditor();
+
+		// create a Navigation Menu called "Test Menu" using the REST API helpers
+		const createdMenu = await requestUtils.createNavigationMenu( {
+			title: 'Primary Menu',
+			content:
+				'<!-- wp:navigation-link {"label":"WordPress","type":"custom","url":"http://www.wordpress.org/","kind":"custom"} /-->',
+		} );
+
+		// Add another so we get a list of Navigation menus in the editor.
+		await requestUtils.createNavigationMenu( {
+			title: 'Another One',
+			content:
+				'<!-- wp:navigation-link {"label":"Another Item","type":"custom","url":"http://www.wordpress.org/","kind":"custom"} /-->',
+		} );
+
+		const editorSidebar = page.getByRole( 'region', {
+			name: 'Navigation',
+		} );
+
+		await editorSidebar
+			.getByRole( 'button', {
+				name: 'Navigation',
+			} )
+			.click();
+
+		// Wait for list of Navigations to appear.
+		editorSidebar.getByRole( 'heading', {
+			name: 'Navigation',
+			level: 1,
+		} );
```

Comment:
what is this for?

Reply:
`getByRole` returns a locator instance, which is lazily evaluated only when performing actions or assertions. In other words, this line doesn't really wait for anything here.

What we could do is to add the `toBeVisible()` assertion to make this explicit, then the comment above won't be needed anymore too.

Reply:
This is an error and thus not intentional.

Reply:
✅